### PR TITLE
dev/java-8 pull request

### PR DESCRIPTION
**Pull request by stevesalas**
- Updates ASM usage for Java 8 support
- Updates database usage for version 1.4.196 support
- Includes workaround for handling errors in processing stage

### CODE REVIEW

File: codepulse/src/main/scala/com/secdec/codepulse/data/model/slick/SlickH2ProjectDataProvider.scala [hunk 1]
```diff
@@ -42,10 +42,10 @@ class SlickH2ProjectDataProvider(folder: File, actorSystem: ActorSystem) extends
 	private val cache = collection.mutable.Map.empty[ProjectId, SlickProjectData]
 
 	private object ProjectFilename {
-		def apply(projectId: ProjectId) = s"${getDbName(projectId)}.h2.db"
```
Comment (baffles) [inline codepulse/src/main/scala/com/secdec/codepulse/data/model/slick/SlickH2ProjectDataProvider.scala]: Why was the filename changed? Is this related to Java 8 support?

My concern is: if we're changing the database filename, this will affect backward compatibility with existing data. We should probably try to not break that.

File: codepulse/src/main/scala/com/secdec/codepulse/tracer/ProjectFileUploadHandler.scala [hunk 2]
```diff
@@ -22,6 +22,8 @@ package com.secdec.codepulse.tracer
 import java.io.File
 import net.liftweb.http.BadResponse
 import net.liftweb.http.LiftResponse
+import net.liftweb.http.RedirectWithState
```
Comment (baffles) [inline codepulse/src/main/scala/com/secdec/codepulse/tracer/ProjectFileUploadHandler.scala]: nitpick: spuriously added imports?

File: codepulse/src/main/scala/com/secdec/codepulse/data/model/slick/SlickH2ProjectDataProvider.scala (hunk 1 above)
Comment (stevesalas) [reply codepulse/src/main/scala/com/secdec/codepulse/data/model/slick/SlickH2ProjectDataProvider.scala]: I share your concern about backward compatibility. The newer version of the H2 driver is unrelated to Java 8 support. It is a dependency of dependency-check version 3.0.0.2, which was committed on 11/16, a day before Code Pulse version 1.1.15 got released.

File: codepulse/src/main/scala/com/secdec/codepulse/tracer/ProjectFileUploadHandler.scala (hunk 2 above)
Comment (stevesalas) [reply codepulse/src/main/scala/com/secdec/codepulse/tracer/ProjectFileUploadHandler.scala]: I removed the two unnecessary import statements from ProjectFileUploadHandler.scala.

**Comment by baffles** [comment]
I didn't test any of this, but the code looks fine to me. 👍
